src/posit/connect/tags.py

```diff
@@ -18,6 +18,36 @@ def find(self) -> list[Tag]:
         pass
 
 
+def _update_parent_kwargs(kwargs: dict) -> dict:
+    """
+    Sets the `parent_id` key in the kwargs if `parent` is provided.
+
+    Asserts that the `parent=` and `parent_id=` keys are not both provided.
+    """
+    parent = kwargs.get("parent", None)
+    if parent is None:
+        # No parent to upgrade, return the kwargs as is
+        return kwargs
+
+    if not isinstance(parent, Tag):
+        raise TypeError(
+            "`parent=` must be a Tag instance. If using a string, please use `parent_id=`"
+        )
+
+    parent_id = kwargs.get("parent_id", None)
+    if parent_id:
+        raise ValueError("Cannot provide both `parent=` and `parent_id=`")
+
+    ret_kwargs = {**kwargs}
+
+    # Remove `parent` from ret_kwargs
+    # and store the `parent_id` in the ret_kwargs below
+    del ret_kwargs["parent"]
+
+    ret_kwargs["parent_id"] = parent["id"]
+    return ret_kwargs
+
+
```

Comment:
Is this logic used elsewhere? If not, should it just be part of the update method implementation?

Reply:
Yes. It is used in
* `Tag.update()`
* `Tags.find()`
* `Tags.create()`